src/lib/types.ts

```diff
@@ -76,6 +77,16 @@ export interface Saga<State, Payload> {
 
 export type AnySaga = Saga<any, any>;
 
+export interface SagaMiddleware {
+  middleware: Middleware;
+  sagaCompletion: () => Promise<void>;
+}
+
+export interface AwaitingAction {
```

Comment:
I would've types this as `AwaitingAction<Payload>`, to make the connection between the `actionCreator` and `resolve` explicit, but I guess on the usage side it would always be `AwaitingAction<any>` then…

Reply:
In the middleware, we would then need to use `AwaitingAction<any>[]` which doesn't give us more type safety. The type-safety is built into the environment, so I don't think we need this.